rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1199,3 +1348,140 @@ def get_basic_contract_info(self, address: ChecksumEvmAddress) -> Dict[str, Any]
+        )
+        registry_coins_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='get_coins',
+            all_arguments=pool_addresses,
+            decode_result=True,
+        )
+        registry_underlying_coins_result = query_single_contract_multicall(
+            ethereum_manager=self,
+            contract=registry_contract,
+            method_name='get_underlying_coins',
+            all_arguments=pool_addresses,
+            decode_result=True,
+        )
+        pools_mapping: CURVE_POOLS_MAPPING_TYPE = {}
+        results_zip = zip(pool_addresses, registry_lp_tokens_result, registry_coins_result, registry_underlying_coins_result)  # noqa: E501
+        for pool_addr, lp_token_encoded, decoded_pool_coins, decoded_pool_underlying_coins in results_zip:  # noqa: E501
+            decoded_lp_token = hex_or_bytes_to_address(lp_token_encoded)
+            pool_coins = []
+            pool_underlying_coins = []
+            for pool_coin in decoded_pool_coins:
+                checksumed_coin_addr = to_checksum_address(pool_coin)
+                if checksumed_coin_addr == ZERO_ADDRESS:
+                    # curve returns array of fixed length. So if current pool utilizes fewer
+                    # coins than the array's length, first values in the result are the real
+                    # coins and all addresses after are just 0x00..00
+                    break
+                pool_coins.append(checksumed_coin_addr)
+            for pool_underlying_coin in decoded_pool_underlying_coins:
+                checksumed_underlying_coin_addr = to_checksum_address(pool_underlying_coin)
+                if checksumed_underlying_coin_addr == ZERO_ADDRESS:
+                    break
+                pool_underlying_coins.append(checksumed_underlying_coin_addr)
+
+            pools_mapping[decoded_lp_token] = (pool_addr, pool_coins, pool_underlying_coins)
+
+        ensure_curve_tokens_existence(ethereum_manager=self, pools_mapping=pools_mapping)
+        save_curve_pools_to_cache(write_cursor=write_cursor, pools_mapping=pools_mapping)
+
+    def update_curve_metapools_cache(self, write_cursor: DBCursor) -> None:
+        """Query pools from curve metapool factory.
+        May raise:
+        - RemoteError if failed to query etherscan or node
+        - NotERC20Conformant if failed to query info while calling get_or_create_evm_token
+        """
+        factory_address_result = CURVE_ADDRESS_PROVIDER.call(
+            ethereum=self,
+            method_name='get_address',
+            arguments=[3],  # 3 is metapool factory id
+        )
+        factory_address = to_checksum_address(factory_address_result)
+        factory_contract = EthereumContract(
+            address=factory_address,
+            abi=CURVE_METAPOOL_FACTORY_ABI,
+            deployed_block=0,
```

Comment:
The problem is that it would require a chain query since this contract is a subject to change. Or is extra chain query here fine?

Reply:
Hmm no it would not just require a chain address. Finding a deploying block number is not easy. But I thought it's a constant factory. If it's not constant it's fine.

You can easily find the current factory and maybe put its deploed block here, but anyway since this is not used it's fine.

But please add a comment whenever you have this `=0` specifying that you don't use the deployed block here. (It's used to limit event queries if you do them).